src/main/java/reposense/parser/ArgsParser.java

```diff
@@ -0,0 +1,91 @@
+    private static final String PROGRAM_DESCRIPTION =
+            "RepoSense is a contribution analysis tool for Git repositories.\n";
+    private static final String MESSAGE_SINCE_DATE_LATER_THAN_UNTIL_DATE =
+            "\"Since Date\" cannot be later than \"Until Date\"";
+
+    private static ArgumentParser getArgumentParser() {
+        ArgumentParser parser = ArgumentParsers
+                .newFor(PROGRAM_USAGE)
+                .build()
+                .defaultHelp(true)
+                .description(PROGRAM_DESCRIPTION);
+
+        parser.addArgument("-config")
+                .required(true)
+                .type(Arguments.fileType().verifyExists().verifyIsFile().verifyCanRead())
+                .metavar("PATH")
+                .help("The CSV File  to the CSV config file");
+
+        parser.addArgument("-output")
+                .metavar("PATH")
+                .type(Arguments.fileType().verifyExists().verifyIsDirectory().verifyCanWrite())
+                .setDefault(new File("."))
+                .help("Directory to output the results");
+
+        MutuallyExclusiveGroup dateGroup = parser.addMutuallyExclusiveGroup();
+
+        dateGroup.addArgument("-since")
+                .metavar("dd/MM/yyyy")
+                .type(new DateArgumentType())
+                .setDefault(Optional.empty())
+                .help("Starting date to filter results");
+
+        dateGroup.addArgument("-until")
+                .metavar("dd/MM/yyyy")
+                .type(new DateArgumentType())
+                .setDefault(Optional.empty())
+                .help("Date to filter results until");
+
+        return parser;
+    }
+
+    public static CliArguments parse(String[] args) throws IllegalArgumentException {
+        CliArguments cliArguments = null;
+        ArgumentParser parser = getArgumentParser();
+
+        try {
+            Namespace results = parser.parseArgs(args);
+
+            File configFile = results.get("config");
+            File outputFile = results.get("output");
+            Optional<Date> sinceDate = results.get("since");
+            Optional<Date> untilDate = results.get("until");
+
+            verifyDatesAreInCorrectRange(sinceDate, untilDate);
+
+            cliArguments = new CliArguments(configFile, outputFile, sinceDate, untilDate);
+        } catch (ArgumentParserException apex) {
+            parser.handleError(apex);
+            System.exit(1);
+        }
+
+        return cliArguments;
+    }
+
+    /**
+     * Verifies that {@code sinceDate} is earlier than {@code untilDate}
+     *
+     * @throws IllegalArgumentException If user-supplied values are invalid
+     */
+    private static void verifyDatesAreInCorrectRange(Optional<Date> sinceDate, Optional<Date> untilDate) {
+        if (sinceDate.isPresent() && untilDate.isPresent() && sinceDate.get().getTime() > untilDate.get().getTime()) {
+            throw new IllegalArgumentException(MESSAGE_SINCE_DATE_LATER_THAN_UNTIL_DATE);
+        }
```

Comment:
should split out the error, just do a `sinceDate.get()`, will throw its own error.
`MESSAGE_SINICE_DATE_LATER_THAN_UNTIL_DATE` not really describe when sinceDate and untilDate is empty.